### ISSUE TITLE
Type error with `modifier`?

### ISSUE THREAD

**Issue by timhwang21** [styled-components/jest-styled-components#268]
Tested on both v6 latest and v7 beta.

The docs show this example:

```typescript
test('nested buttons are flexed', () => {
  const tree = renderer.create(<ButtonList><Button /></ButtonList>).toJSON()
  expect(tree).toHaveStyleRule('flex', '1 0 auto', {
    modifier: css`${Button}`,
  })
})
```

However, when using Typescript, we get the following error: `Type 'FlattenSimpleInterpolation' is not assignable to type 'string'.`

Seems like a simple enough fix that I can tackle if needed (change interface for `modifier`), but just wanted to make sure that the docs actually describe accurate + intended behavior before I start diving into this. Thanks!

**Comment by acfasj** [comment]
come across same issue

**Comment by jarvelov** [comment]
Hate to be another "me too" but this issue presented itself for me as well. Right now I'm casting it as a string for now, but it would be nice if the types can be updated.

**Comment by MateusGruber** [comment]
I've used `any` to trespass this issue 😔

```javascript
test('nested buttons are flexed', () => {
  const tree = renderer.create(<ButtonList><Button /></ButtonList>).toJSON()
  expect(tree).toHaveStyleRule('flex', '1 0 auto', {
    modifier: css`${Button}` as any,
  })
})
```

**Comment by cscheffauer** [comment]
faced the same issue - any updates on this?

**Comment by timhwang21** [comment]
I don't really use this library any more but still get emails on this issue 😄 Would appreciate it anyone could verify if my fix at #367 resolves this issue.

**Comment by cscheffauer** [comment]
@timhwang21 approved it :) unfortunately i don't have the necessary rights to merge this PR.
